### DISCUSSION TITLE
Conversational Retrieval Agent > how to return the document nr/paragraph of the answer?

### DISCUSSION THREAD

**Discussion by GiovanniDeBeyter** [FlowiseAI/Flowise#999] [Q&A]
Hi, 

I have a Conversational Retrieval Agent, working with documents from PineCone, and a memory.

As a functional test, I ask specific questions which I know the answers are in de documents. (there are several docs in the store).

Can I add a code snippet or change a setting so the chatbot adds together with the answer also the reference of the page / paragraph and source document where it got its answer from? for example 'car insurance policy, page 2, paragraph 3'? 

I know there are some Chain have the switch 'return documents' but i don't see it in Agents.

**Comment by HenryHengZJ** [comment]
in agent, its still not possible to return source documents yet, but we are looking to support that in future
